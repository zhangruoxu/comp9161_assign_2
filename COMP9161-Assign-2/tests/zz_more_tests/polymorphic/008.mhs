main = letfun f x = (if (fst x) then (fst x) else (snd x));

