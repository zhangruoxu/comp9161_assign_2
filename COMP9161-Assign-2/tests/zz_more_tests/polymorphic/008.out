main :: ((Bool * Bool) -> Bool) = (letfun f :: ((Bool * Bool) -> Bool) x = if fst x
                                                                              then fst x
                                                                              else snd x);
